Sources/Topsort-Analytics/AuctionManager.swift

```diff
@@ -0,0 +1,50 @@
+import Foundation
+
+private let AUCTIONS_TOPSORT_URL = URL(string: "https://api.topsort.com/v2/auctions")!
+
+class AuctionManager {
+    public static let shared = AuctionManager()
+    private init() {
+        self.client = HTTPClient(apiKey: nil)
+    }
+    private var url: URL = AUCTIONS_TOPSORT_URL
+    private var client: HTTPClient
+
+    public func configure(apiKey: String, url: String?) {
+        self.client.apiKey = apiKey
+        if let url = url {
+            guard let url = URL(string: "\(url)/auctions") else {
+                fatalError("Invalid URL")
+            }
+            self.url = url
+        }
+    }
+
+    public func executeAuctions(auctions: [Auction]) async -> AuctionResponse? {
+        var response: AuctionResponse?
+        guard let auctionsData = try? JSONEncoder().encode(["auctions": auctions]) else {
+            print("failed to serialize auctions: \(auctions)")
+            return nil
+        }
+
+        let id = UUID()
+        do {
+            let resultData = try await self.client.asyncPost(url: self.url, data: auctionsData)
+            response = self.process_response(id: id, result: .success(resultData))
+        } catch {
+            print("Error posting auctions: \(error)")
+        }
+        return response
+    }
+
+    private func process_response(id: UUID, result: Result<Data?, HTTPClientError>) -> AuctionResponse? {
+        switch result {
+            case .success(let data):
+                let resultData = try? JSONDecoder().decode(AuctionResponse.self, from: data ?? Data())
```

Comment:
If `data` is `nil` we don't want to be decoding from an empty Data. Better to throw in that case.

Reply:
I'd also refactor de decoding/encoding away into functions that you can easily test